Problem 5 - Map/map.js

```diff
@@ -0,0 +1,24 @@
+var iTechArt = iTechArt || {};
+iTechArt.training = iTechArt.training || {};
+iTechArt.training.utils = iTechArt.training.utils || {};
+
+/**
+ *  Implement a function that creates new array with the results of calling a provided function
+ *  on every element in array
+ */
+
+
+iTechArt.training.utils.map = function(array, fn) {
+    var resultArray = [];
+    for (var i = 0, arrayLength = array.length; i < arrayLength; i++){
+        if (i in array){
+            // according to original map function it is a known practice
```

Comment:
Map itself originates to a functional paradigm. Basically it is operation on  [Monad](http://en.wikipedia.org/wiki/Monad_%28functional_programming%29). Thus passing parameters you specified, binds it to very specific case. It can expect something like this, but it shouldn't - not a known practice